src/server/scripts/Kalimdor/zone_desolace.cpp

```diff
@@ -88,12 +88,20 @@ class npc_aged_dying_ancient_kodo : public CreatureScript
                     DoCast(me, SPELL_KODO_KOMBO_DESPAWN_BUFF, true);
 
                     me->UpdateEntry(NPC_TAMED_KODO);
+                    me->CombatStop();
+                    me->DeleteThreatList();
+                    me->SetSpeed(MOVE_RUN, 0.6f, true);
                     me->GetMotionMaster()->MoveFollow(caster, PET_FOLLOW_DIST, me->GetFollowAngle());
+                    me->setActive(true);
```

Comment:
Well, @MitchesD said that I should leave it in, because without it, the kodo will stop following if you move out of sight. According to retail comments on wowhead, the kodo should follow even if player runs or rides ahead out of visibility.